rust/zypp-agama/src/lib.rs

```diff
@@ -72,9 +75,13 @@ where
         let mut closure = progress;
         let cb = get_progress_callback(&closure);
         let c_root = CString::new(root).unwrap();
-        let mut status: Status = Status { state: Status_STATE_STATE_SUCCEED, error: null_mut() };
-        let status_ptr = &mut status as *mut _ as *mut Status;
-        zypp_agama_sys::init_target(c_root.as_ptr(), status_ptr, cb, &mut closure as *mut _ as *mut c_void);
+        let (status, status_ptr) = create_status_with_pointer();
```

Comment:
The problem with this is that `status` will be copied(?) and live at a new address, while `status_ptr` will point to the old copy.
For now, you can revert to your old copy-paste Status code while I debug a working replacement

Reply:
good catch. I thought that rust copy is optimized to not do move, but we should not depends on it and probably in test it do not optimize that unnecessary copy of structs

Reply:
of course option can be some macro that will just expand to generate code, but probably overkill in this phase of development

Reply:
I still do not understand what happens and why, will make a minimal test for it